cli/operator/node.go

```diff
@@ -338,6 +338,35 @@ var StartNodeCmd = &cobra.Command{
 			nodeProber.AddNode("event syncer", eventSyncer)
 		}
 
+		// Increase MaxPeers if the operator is subscribed to many subnets.
+		// TODO: use OperatorCommittees when it's fixed.
+		const (
+			baseMaxPeers   = 60
+			maxPeersLimit  = 150
+			peersPerSubnet = 3
+		)
+		start := time.Now()
+		myValidators := nodeStorage.ValidatorStore().OperatorValidators(operatorData.ID)
+		mySubnets := make(records.Subnets, networkcommons.SubnetsCount)
+		myActiveSubnets := 0
+		for _, v := range myValidators {
+			subnet := networkcommons.CommitteeSubnet(v.CommitteeID())
+			if mySubnets[subnet] == 0 {
+				mySubnets[subnet] = 1
+				myActiveSubnets++
+			}
+		}
+		idealMaxPeers := min(baseMaxPeers+peersPerSubnet*myActiveSubnets, maxPeersLimit)
+		if cfg.P2pNetworkConfig.MaxPeers < idealMaxPeers {
+			logger.Warn("increasing MaxPeers to match the operator's subscribed subnets",
+				zap.Int("old_max_peers", cfg.P2pNetworkConfig.MaxPeers),
+				zap.Int("new_max_peers", idealMaxPeers),
+				zap.Int("subscribed_subnets", myActiveSubnets),
+				zap.Duration("took", time.Since(start)),
+			)
+			cfg.P2pNetworkConfig.MaxPeers = idealMaxPeers
+		}
```

Comment:
Is there a need to even have `cfg.P2pNetworkConfig.MaxPeers` value now if we are gonna overwrite it in most cases anyway ? Perhaps we want to remove it to keep it simple (otherwise later on we/somebody might wrongly think the `MaxPeers` value he sets in config file is the value that's gonna be used - which is not gonna be the case).

Reply:
this is a temporary solution and we wouldnt want to educate everyone about a new configuration only to take it down very soon after

Reply:
although we could potentially rename it to TargetPeers and just leave it as is even if we end up reverting this PR

Reply:
@iurii-ssv i made some changes to make it more clear that MaxPeers is currently not the max